Sources/Sentry/SentryClient.m

```diff
@@ -641,7 +641,13 @@ - (void)setSdk:(SentryEvent *)event
     id integrations = event.extra[@"__sentry_sdk_integrations"];
     if (!integrations) {
         integrations = [NSMutableArray new];
-        for (NSString *integration in SentrySDK.currentHub.installedIntegrationNames) {
+
+        // The SDK can send an event before it installs all integrations, which add to this list. To
+        // avoid an exception while iterating over a list while items get added to it, we copy it.
+        // With that approach, we might not have the complete integration list. To fix this, the SDK
+        // would need to ask all integrations first if they should be installed before actually
+        // installing them, which would be quite a bit a of a refactoring.
+        for (NSString *integration in SentrySDK.currentHub.installedIntegrationNames.copy) {
```

Comment:
h: `copy` is not thread safe.

This is the test I used to check this:
```objc
    __block BOOL Finished = false;
    dispatch_group_t group = dispatch_group_create();
    
    NSMutableArray* array = [[NSMutableArray alloc] init];
    
    dispatch_queue_t firstQueue = dispatch_queue_create("add queue", nil);
    dispatch_queue_t secondQueue = dispatch_queue_create("list queue", nil);
    dispatch_group_async(group, firstQueue, ^{
        for (int i = 0; i<100000; i++) {
            [array addObject:[NSNumber numberWithInt:i]];
        }
        Finished = true;
    });
    
    dispatch_async(secondQueue, ^{
        while (!Finished) {
            NSArray * copy = array.copy;
            NSLog(@"%lu", copy.count);
        }
    });
    
    dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
```
It crashes every time.
It's also possible that my test is not good enough 😅

Reply:
Make sense, I will add a thread safe copy, or I'm going to refactor the code, as pointed out by @kevinrenskers.